server/src/main/java/org/elasticsearch/plugins/UberModuleClassLoader.java

```diff
@@ -66,7 +67,30 @@ static UberModuleClassLoader getInstance(ClassLoader parent, String moduleName,
     @SuppressForbidden(reason = "need access to the jar file")
     @SuppressWarnings("removal")
     static UberModuleClassLoader getInstance(ClassLoader parent, String moduleName, List<Path> jarPaths, Set<String> moduleDenyList) {
-        ModuleFinder finder = ModuleSupport.ofSyntheticPluginModule(moduleName, jarPaths.toArray(new Path[0]), Set.of());
+        ModuleLayer mparent = ModuleLayer.boot();
+
+        // Every module requires java.base by default, but we can add all other modules
+        // that are not in the deny list so that plugins can use, for example, java.management
+        Set<String> requires = mparent.modules()
+            .stream()
+            .filter(Module::isNamed)
+            .map(Module::getName)
+            .filter(name -> "java.base".equals(name) == false)
+            .filter(name -> moduleDenyList.contains(name) == false)
+            .collect(Collectors.toSet());
+        // We also need to gather the service names in the public Java API and mark our
+        // synthetic plugin as using them, since we don't know which services our jars
```

Comment:
I don't know of any reason we'd want to deny access to a specific service in the platform. It would need to be a case where we want to allow access to some packages in the module but deny access to some service it provides, because otherwise we could just use the deny-list.

For what it's worth, here are the services I'm finding in my environment:
```
com.sun.jdi.connect.Connector
com.sun.tools.attach.spi.AttachProvider
java.nio.file.spi.FileSystemProvider
java.util.random.RandomGenerator
javax.print.PrintServiceLookup
javax.print.StreamPrintServiceFactory
javax.sound.midi.spi.MidiDeviceProvider
javax.sound.midi.spi.MidiFileReader
javax.sound.midi.spi.MidiFileWriter
javax.sound.midi.spi.SoundbankReader
javax.sound.sampled.spi.AudioFileReader
javax.sound.sampled.spi.AudioFileWriter
javax.sound.sampled.spi.FormatConversionProvider
javax.sound.sampled.spi.MixerProvider
jdk.jshell.spi.ExecutionControlProvider
```